src/main/java/com/squareup/javapoet/CodeWriter.java

```diff
@@ -449,21 +448,28 @@ CodeWriter emitAndIndent(String s) throws IOException {
       if (trailingNewline) {
         emitIndentation();
         if (javadoc) {
-          out.append(" * ");
+          append(" * ");
         } else if (comment) {
-          out.append("// ");
+          append("// ");
         }
       }
-
-      out.append(line);
+      append(line);
       trailingNewline = false;
     }
     return this;
   }
 
-  private void emitIndentation() throws IOException {
+  private void emitIndentation() {
     for (int j = 0; j < indentLevel; j++) {
-      out.append(indent);
+      append(indent);
+    }
+  }
+
+  private void append(CharSequence csq) {
+    try {
+      out.append(csq);
+    } catch (IOException e) {
```

Comment:
No, the signatures of `JavaFile.writeTo` methods are not affected. All four still declare and throw `IOException`s if there are, for example file system related, i/o problems.

For those exceptions caught here, I could wrap them in an unchecked exception, that can be analyzed by `JavaFile.writeTo(Appendable)` for re-throwing the cause. That'd preserve the original behaviour in all cases - and leaves the `toString()` implementation `IOException`-free.